packages/airnode-validator/src/ois/ois.test.ts

```diff
@@ -164,3 +146,90 @@ it('fails if apiSpecifications.security.<securitySchemeName> is not defined in a
     ])
   );
 });
+
+describe('apiSpecification parameters validation', () => {
+  it('fails if "apiSpecification.paths" parameter is not defined in "endpoints"', () => {
+    const invalidOis = loadOisFixture();
+    invalidOis.apiSpecifications.paths['/convert'].get!.parameters.push({
+      in: 'query',
+      name: 'non-existing-parameter',
+    });
+
+    expect(() => oisSchema.parse(invalidOis)).toThrow(
+      new ZodError([
+        {
+          code: 'custom',
+          message: 'Parameter "non-existing-parameter" not found in "fixedOperationParameters" nor "parameters"',
+          path: ['ois', 'endpoints', 0],
+        },
+      ])
+    );
+  });
+
+  it('"endpoint" parameter must reference parameter from "apiSpecification.paths"', () => {
+    const invalidOis = loadOisFixture();
+    invalidOis.endpoints[0].parameters.push({
+      name: 'some-new-param',
+      default: 'EUR',
+      operationParameter: {
+        in: 'query',
+        name: 'non-existing-param',
+      },
+    });
+
+    expect(() => oisSchema.parse(invalidOis)).toThrow(
+      new ZodError([
+        {
+          code: 'custom',
+          message: 'No matching API specification parameter found in "apiSpecifications" section',
+          path: ['ois', 'endpoints', 0, 'parameters', 3],
+        },
+      ])
+    );
+  });
+
+  it('handles multiple endpoints for the same API specification', () => {
+    const ois = loadOisFixture();
+    ois.endpoints.push(cloneDeep(ois.endpoints[0]));
+    ois.apiSpecifications.paths['/convert'].get!.parameters.push({
+      in: 'query',
+      name: 'api-param-name',
+    });
+
+    expect(() => oisSchema.parse(ois)).toThrow(
+      new ZodError([
+        {
+          code: 'custom',
+          message: 'Parameter "api-param-name" not found in "fixedOperationParameters" nor "parameters"',
```

Comment:
If you decide to change the spelling:
```suggestion
          message: 'Parameter "api-param-name" not found in "fixedOperationParameters" or "parameters"',
```